### PR TITLE
Enhancement/5832 deb rpm docker tests

### LINKED ISSUE #5832: Prevent users from upgrading standalone agents in container, deb, and rpm environments
**Describe the enhancement:**

Standalone Elastic Agents that are running in a container or deployed via DEB or RPM packages should not be upgradable using the `elastic-agent upgrade ...` CLI command.

**Describe a specific use case for the enhancement or feature:**

**What is the definition of done?**
Running `elastic-agent upgrade` for standalone Elastic Agents should fail with a descriptive message when Elastic Agent is deployed:
* [ ] in a container
* [ ] via a DEB package
* [ ] via an RPM package

### PR DESCRIPTION
**Pull request by kaanyalti**
- Enhancement
## What does this PR do?
Adds tests validating elastic-agent upgrade command does not and should not work for deb, rpm and docker.

## Why is it important?
We currently don't have any tests validating that the upgrade command does not work

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- [x] I have added an integration test or an E2E test

## How to test this PR locally

- Build deb, rpm packages and docker images
- Run the following commands
```
 TEST_PACKAGES='deb' SNAPSHOT=true TEST_GROUPS="deb" TEST_PLATFORMS="linux/amd64" GOTEST_FLAGS='-test.run ^TestRestrictUpgradeDeb$' mage integration:test
```

```
 TEST_PACKAGES='rpm' SNAPSHOT=true TEST_GROUPS="rpm" TEST_PLATFORMS="linux/amd64" GOTEST_FLAGS='-test.run ^TestRestrictUpgradeRPM$' mage integration:test
```

```
K8S_VERSION=v1.31.0 TEST_INTEG_CLEAN_ON_EXIT=true INSTANCE_PROVISIONER=kind STACK_PROVISIONER=stateful TEST_PLATFORMS="kubernetes/arm64/1.31.0/<IMAGE TYPE>" SNAPSHOT=true mage integration:kubernetes
```
The image type can be `basic`, `wolfi` or any other listed in the test itself

## Related issues

- Closes #5832

### CODE REVIEW

**Comment by rowlandgeoff** [comment]
Maybe mark this PR as Draft for now while you keep testing it? I really wouldn't want to accidentally commit a change where the Buildkite pipelines are all getting commented out. 😉

**Comment by rowlandgeoff** [comment]
> trying to run only my test in ci

I don't think we currently have an easy mechanism of doing this. Perhaps something we can consider in the future.

**Comment by kaanyalti** [comment]
buildkite test this

File: testing/integration/kubernetes_agent_standalone_test.go [hunk 1]
```diff
@@ -479,6 +480,215 @@ func TestKubernetesAgentHelm(t *testing.T) {
...
+					t.Logf("stdout: %s\n", statusString)
+					t.Logf("stderr: %s\n", stderr.String())
+					t.FailNow()
+					return
+				}
+
+				stdout.Reset()
+				stderr.Reset()
+
+				upgradeCmd := []string{"elastic-agent", "upgrade", "1.0.0"}
```
Comment (pkoutsovasilis) [inline testing/integration/kubernetes_agent_standalone_test.go:674]: sorry, but is it too much to ask to align this test with the new structure of k8s integration tests? e.g. define a k8sTestStep that checks that the upgrade cmd fails, and just add this step in one of the existing k8s tests?! 🙂

Comment (kaanyalti) [reply testing/integration/kubernetes_agent_standalone_test.go:674]: Looking into it right away

**Comment by kaanyalti** [comment]
> the problem with integration test is that `/etc/elastic-agent` is kept there and new installation assumes enrolled agent. when you run purge it will be removed and next install will work. why purge works and uninstall does not is this sneaky condition we have [here](https://github.com/elastic/elastic-agent/blob/a5eb77b2440d037cf3e9d627bc3da0b2cc49d008/dev-tools/packaging/templates/linux/postrm.sh.tmpl#L7):
> 
> ```shell
> case "$1" in
>   purge)
>     rm -rf /var/lib/elastic-agent /var/log/elastic-agent /etc/elastic-agent
>     ;;
> 
>   # 0 is for rpm uninstall
>   upgrade|remove|failed-upgrade|abort-install|abort-upgrade|disappear|0)
> ```
> 
> what we need to do is move `rm -rf /etc/elastic-agent` outside of this this if statement. this can be troublesome in case of upgrades. ideally we remove not only on purge but also uninstall but not upgrade
> 
> behavior also described in PR #4334

Based on our discussion implement manual removal of agent files for RPM clean up

**Review by michalpristas — COMMENTED** [review]
i like that

**Review by michalpristas — APPROVED** [review]

**Review by pkoutsovasilis — APPROVED** [review]
LGTM. Nice work 🙂

**Pull request merged**